### PR TITLE
addresses inconsistencies when using `sane generate [singular noun]`

### PR DESCRIPTION
**Pull request by jamemackson**
This PR is to address https://github.com/artificialio/sane/issues/123 where we get inconsistent results when calling `sane generate user` vs `sane generate users`.  With these changes, both calls should result in the same modeling being created.  We're basically pluralizing the name passed in when passing the value to the respective ember and sails CLI's.

### CODE REVIEW

**Comment by Globegitter** [comment]
This just pluralises the resource no matter what. I think what would be better behaviour (to make it consistent between ember and sails) is to just `singularize` whatever parameter is being passed to `sails`. 

Imo you should still be able to run `sane generate resource friend` and let ember create all the singular stuff. The main issue seems to be when you run `sane generate resource friends` that sails just creates a `Friends` model and controller rather than a `Friend` one.

**Comment by mgenev** [comment]
Yeah, sane generator should be smart so that it tells sails - > singular, ember -> plural, that's about it, right?

**Comment by Globegitter** [comment]
Yep @mgenev much more concise :P

**Comment by jamemackson** [comment]
I apologize, I thought we wanted the sails side to be plural.  A simple change to singularize that.  uno momento :)

**Comment by jamemackson** [comment]
and just so we're clear here, 
* we want the sails model and controller to be singular
* we want the ember model to be singular
* we want the ember route and template to be whatever the user passed in

would you all agree?

**Comment by Globegitter** [comment]
@jamemackson ember is clever enough to do the right thing itself. So we just have to help out sails here :)

**Comment by jamemackson** [comment]
@Globegitter thanks, I'll leave the ember side alone then and just help sails out :)

**Comment by jamemackson** [comment]
@Globegitter @mgenev see how this looks now.  only passing pluralized name to sails; straight through to ember.

**Comment by Globegitter** [comment]
@jamemackson One little trick, you can either do `var singularize = require('pluralize').singular;` to directly require the one function we need or use new ES6 syntax and let JS do it: `var { singular } = require('pluralize');`. Then it is good to merge ;)

**Comment by jamemackson** [comment]
sure, I can handle that. :)

**Comment by IanVS** [comment]
Can you try squashing these commits?  I know `rebase` is your favorite, but it should be pretty straight forward:

```
$ git rebase -i HEAD ~3
// leave the top line as-is, on the second and third lines, change "pick" to "s", then save

$ git push -f origin master
```

**Comment by jamemackson** [comment]
@IanVS done.  Hopefully done correctly lol

**Comment by IanVS** [comment]
Hm, close, but it's still showing two commits.  Try again with `git rebase -i HEAD~2` (I think for some reason there was an extra space between "HEAD" and the "~" in my previous comment, I corrected now.)

**Comment by jamemackson** [comment]
yea I saw an error when I tried it with the space so I ran it without and it worked... I'll give it another go here quick and see if that fixes it up.

**Comment by jamemackson** [comment]
ok now it looks like it's squashed down to 1 commit... I'll be back in a couple hours if I need to do more there...

**Comment by IanVS** [comment]
I think it would be good to add acceptance tests to verify Sails and Ember behave as expected. You can take a look at the current test at: https://github.com/artificialio/sane/blob/master/tests/acceptance/generateTest.js#L59-60 to get started, but I think we should think about refactoring slightly so as not to run generate over and over with the same arguments. One way would be to extract the generate out into a beforeEach, and maybe add separate describes to group the tests by the generate comma d that needs to be run.

**Comment by Globegitter** [comment]
I was just thinking the same thing today (about not generating it over and over again). Yeah that is a good call @IanVS. Let's use this chance to also improve the test situation a bit!

**Comment by mgenev** [comment]
Sounds like a new story/PR to me? We can let @jamemackson off the hook here?

**Comment by Globegitter** [comment]
@mgenev The minor refactor yes - if @jamemackson is up for it though it would be great to have some acceptance tests that ensure that the singularization works correctly (by checking the created file names)

**Comment by IanVS** [comment]
I agree refactoring could/should be a separate activity.  For now as part of this pull request I think we need tests which:

- Assert that the proper filenames are generated in client when singular format is used (e.g. `sane generate resource user`)
- Assert that the proper filenames are generated in server when singular format is used (e.g. `sane generate resource user`)
- Assert that the proper filenames are generated in client when plural format is used (e.g. `sane generate resource users`)
- Assert that the proper filenames are generated in server when plural format is used (e.g. `sane generate resource users`)

**Comment by jamemackson** [comment]
I am happy to tackle the tests.  Probably not until later in the week though (and I mean it this time haha - I really do need to buckle down on some freelance work for a couple of days, even though it is so much less fun that working on this!) But are there other tests that do something similar I can reference?  If not so big deal, I can figure it out.

**Comment by IanVS** [comment]
@jamemackson, you should be able to use the `assertFile()` helper here, similar to the current `generateTest.js`. It could look something like:

```
await generate('resource user');

  assertFile('client/app/models/user.js');
  assertFile('server/api/models/user.js');
  // etc. . . 
```

**Comment by Globegitter** [comment]
Interesting that appveyor passed and one single travis test failed. @jamemackson any idea why that could be? But let me re-run the tests.

File: tests/acceptance/generateTest.js [hunk 1]
```diff
@@ -57,28 +57,115 @@ describe('Acceptance: sane generate', function () {
...
+  it('should generate a singular sails model  when passed a singular value', async function () {
+    await generate('resource foo');
+
+    assertFile('server/api/models/Foo.js');
+  });
+
+  it('should generate a singular sails model when passed a plural value', async function () {
+    await generate('resource bars');
+
+    assertFile('server/api/models/bar.js');
```
Comment (IanVS) [inline tests/acceptance/generateTest.js]: Sails generates a capitalized model name, so `Bar.js`

Comment (Globegitter) [inline tests/acceptance/generateTest.js]: Ahhh that also explains the difference between Windows and Linux. Good catch,

Comment (jamemackson) [inline tests/acceptance/generateTest.js]: I'll get that fixed up in a bit here when I get back to the computer:)

**Comment by Globegitter** [comment]
Thanks @jamemackson and could you also please add a CHANGELOG.md entry.

**Comment by jamemackson** [comment]
hopefully this foxes things up... :)

**Comment by Globegitter** [comment]
Thank you again @jamemackson and as a first PR acceptance gift we are accepting without an further squashing ;)

**Pull request merged**